BannerlordTwitch/BannerlordTwitch/Api/BLTApi.cs

```diff
@@ -0,0 +1,148 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Dynamic;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading.Tasks;
+using System.Web;
+using BannerlordTwitch;
+using BannerlordTwitch.Util;
+using Newtonsoft.Json;
+using TwitchLib.Client.Models;
+using TwitchLib.Client.Models.Internal;
+
+namespace BannerlordApi
+{
+    public class BLTApi
+    {
+        List<BLTApiUrl> bltApiUrls = new List<BLTApiUrl>();
+        public readonly int port = 9000;
+        private static HttpListener listener;
+        public BLTApi()
+        {
+            _ = RunServerAsync();
+            bltApiUrls.Add(new BLTApiUrl("GET", "", "Show all registered api path", OnApiVisit));
+            bltApiUrls.Add(new BLTApiUrl("GET", "command", "Used to execute commands", OnCommand));
+        }
+
+        public async Task RunServerAsync()
+        {
+            Trace.WriteLine("Api server starting");
+            bool runServer = true;
+            listener = new HttpListener();
+            listener.Prefixes.Add("http://localhost:"+port+"/api/");
+            listener.Start();
+
+            while (runServer)
+            {
+                HttpListenerContext ctx = await listener.GetContextAsync();
+
+                HttpListenerRequest req = ctx.Request;
+                HttpListenerResponse resp = ctx.Response;
+
+                OnUrlRequest(req, resp);
+            }
+        }
+
+        public void OnUrlRequest(HttpListenerRequest req, HttpListenerResponse resp)
+        {
+            Trace.WriteLine("[VISITED]" + req.Url.ToString());
+            bltApiUrls.ForEach(bltApiUrl =>
+            {
+                if (
+                    "http://localhost:"+port+"/api/" + bltApiUrl.url == req.Url.ToString().Split('?')[0] || 
```

Comment:
You should just put them into a Uri and compare with that:
https://docs.microsoft.com/en-us/dotnet/api/system.uri?view=netframework-4.8
It has a Uri.Compare function where you can specify the part to compare, and of course it is semantic compare.